### PR TITLE
Fix/avoid leading zero rate queue

### PR DESCRIPTION
**Pull request by Hany-Almnaem**
## Summary

This PR addresses [#82 (Avoid leading-zero rates in the rate-change queue)](https://github.com/FilOzone/fws-payments/issues/82) by ensuring that, when settling a rail with a zero payment rate and an empty rate change queue, the contract advances `settledUpTo` directly—without enqueuing or modifying the queue.

---

## Changes Made
1-**Optimized Logic in `Payments.sol`**
- Settlement Optimization: 
  Updated the settlement logic to check for the special case where both the rate change queue is empty *and* the payment rate is zero:
- Added a public getter `getRateChangeQueueSize(uint256 railId)` in `Payments.sol`
  Allows test contracts and clients to verify the rate change queue size for any rail.
  
2- **Added a new unit test in `RailSettlement.t.sol`**  
  - Test: `testZeroRateRail_OptimizedSettlement`
  - Verifies that settling a zero-rate rail with an empty queue:
    - Advances `settledUpTo`
    - Does not enqueue or modify the queue (queue size remains zero before and after settlement)
    - Returns the correct optimization note in the result

---

## Rationale

- Prevents redundant zero-rate entries in the queue for rails that start with zero payment rate.
- Aligns with the issue’s goal: **Only move funds or create queue entries when strictly needed**.
- Ensures test coverage for this important edge case.

### CODE REVIEW

File: src/Payments.sol [hunk 1]
```diff
@@ -1003,7 +1003,12 @@ contract Payments is
                 )
             );
         }
-
+        // This avoids redundant zero-rate entries and saves gas.
```
[the remark this answers is not in the document]
Comment (aarshkshah1992) [reply src/Payments.sol:1006]: Issue https://github.com/FilOzone/filecoin-services-payments/issues/82 is about the `enqueueRateChange` function. That function enqueues the current rate on a queue for an arbitrated rail when the operator changes the rate of the rail so that we don't need to settle/arbitrate payment upto the current epoch using the old rate.

We just enqueue the old rate and then settle later when the user calls `settleRail`.

Issue 82 is basically saying that we do not need to enqueue the old rate if the old rate is 0 and the rate change queue is empty. We can just advance the `rail.settleUpto` epoch upto the current epoch instead and consider the rail settled upto the current epoch (because the old rate is 0, there's nothing to pay out here).

**Review by aarshkshah1992 — CHANGES_REQUESTED** [review]
Please take a look at my comments. This issue about not enqueuing rate changes during rail rate modification and not about rail settlement.

**Comment by aarshkshah1992** [comment]
Also @Hany-Almnaem -> please can you rebase on master and solve that conflict ?

**Comment by aarshkshah1992** [comment]
@Hany-Almnaem Please rebase this PR on the latest master before you make the change. Thank You !

**Comment by Hany-Almnaem** [comment]
Thanks for the earlier pointers, @aarshkshah1992 🙏

- I’ve dropped the earlier settlement-layer change and moved the logic entirely into enqueueRateChange (see c48b3f8).

- Now, if the current rate is 0 and the queue is empty, we simply bump  settledUpTo to block.number and return—no queue insert, no extra gas.

- Added getRateChangeQueueSize() for visibility and a focused test testModifyRailPayment_SkipsZeroRateEnqueue() to prove the behaviour.

- All tests are green on the latest master after the rebase. 

Happy to adjust as needed. Otherwise I’ll mark the outdated discussion on the previous approach as resolved.
